predicates-push-down.md

```diff
@@ -0,0 +1,151 @@
+title: Predicates Push Down
+category: reference
+aliases: ['/docs/dev/predicates-push-down.md/','/docs/dev/reference/performance/predicates-push-down/']
+---
+
+# Predicates Push Down(PPD)
+
+This document introduces one of the TiDB's logic optimization rules—Predicate Push Down(PPD). It aims to help readers understand the predicate push down and know its suitable and unsuitable scenarios.
+
+PPD pushes down selection operators as close to data source as possible to complete data filtering as early as possible. Thereby significantly reducing the cost of data transmission or computation. 
+
+## Examples
+
+The following cases describes the optimization of PPD. Examples 1, 2, and 3 are cases where PPD is applicable, and examples 4, 5, and 6 are cases where PPD is not applicable.
+
+### Case 1: push predicates to storage layer
+
+```sql
+create table t(id int primary key, a int);
+explain select * from t where a < 1;
++-------------------------+----------+-----------+---------------+--------------------------------+
+| id                      | estRows  | task      | access object | operator info                  |
++-------------------------+----------+-----------+---------------+--------------------------------+
+| TableReader_7           | 3323.33  | root      |               | data:Selection_6               |
+| └─Selection_6           | 3323.33  | cop[tikv] |               | lt(test.t.a, 1)                |
+|   └─TableFullScan_5     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo |
++-------------------------+----------+-----------+---------------+--------------------------------+
+3 rows in set (0.00 sec)
+```
+
+In this query，pushing down the predicate `a < 1` on TiKV to filter the data can reduce the overhead caused by network transmission.
+
+### Case 1: push predicates to storage layer
+
+```sql
+create table t(id int primary key, a int not null);
+explain select * from t where a < substring('123', 1, 1);
++-------------------------+----------+-----------+---------------+--------------------------------+
+| id                      | estRows  | task      | access object | operator info                  |
++-------------------------+----------+-----------+---------------+--------------------------------+
+| TableReader_7           | 3323.33  | root      |               | data:Selection_6               |
+| └─Selection_6           | 3323.33  | cop[tikv] |               | lt(test.t.a, 1)                |
+|   └─TableFullScan_5     | 10000.00 | cop[tikv] | table:t       | keep order:false, stats:pseudo |
++-------------------------+----------+-----------+---------------+--------------------------------+
+```
+
+This query has the same execution plan as the query in case 1, because the input parameters of the `substring` of the predicate `a < substring('123', 1, 1)` are constants, so they can be calculated in advance. Then simplify to get the equivalent predicate `a < 1`. Further, TiDB can push `a < 1` down to TiKV.
```

Comment:
```suggestion
This query has the same execution plan as the query in case 1, because the input parameters of the `substring` of the predicate `a < substring('123', 1, 1)` are constants, so they can be calculated in advance. Then the predicate is simplified to the equivalent predicate `a < 1`. After that, TiDB can push `a < 1` down to TiKV.
```